src/category_theory/enriched/basic.lean

```diff
@@ -0,0 +1,435 @@
+(id_comp : Π X Y, (λ_ (X ⟶[] Y)).inv ≫ (id X ⊗ 𝟙 _) ≫ comp X X Y = 𝟙 _ . obviously)
+(comp_id : Π X Y, (ρ_ (X ⟶[] Y)).inv ≫ (𝟙 _ ⊗ id Y) ≫ comp X Y Y = 𝟙 _ . obviously)
+(assoc :
+  Π W X Y Z, (α_ _ _ _).inv ≫ (comp W X Y ⊗ 𝟙 _) ≫ comp W Y Z = (𝟙 _ ⊗ comp X Y Z) ≫ comp W X Z
+  . obviously)
+
+notation X ` ⟶[`V`] ` Y:10 := (enriched_category.hom X Y : V)
+
+variables (V) {C : Type u₁} [enriched_category V C]
+
+/--
+The `𝟙_ V`-shaped generalized element giving the identity in a `V`-enriched category.
+-/
+def e_id (X : C) : 𝟙_ V ⟶ (X ⟶[V] X) := enriched_category.id X
+/--
+The composition `V`-morphism for a `V`-enriched category.
+-/
+def e_comp (X Y Z : C) : (X ⟶[V] Y) ⊗ (Y ⟶[V] Z) ⟶ (X ⟶[V] Z) := enriched_category.comp X Y Z
+
+@[simp, reassoc]
+lemma e_id_comp (X Y : C) :
+  (λ_ (X ⟶[V] Y)).inv ≫ (e_id V X ⊗ 𝟙 _) ≫ e_comp V X X Y = 𝟙 (X ⟶[V] Y) :=
+enriched_category.id_comp X Y
+
+@[simp, reassoc]
+lemma e_comp_id (X Y : C) :
+  (ρ_ (X ⟶[V] Y)).inv ≫ (𝟙 _ ⊗ e_id V Y) ≫ e_comp V X Y Y = 𝟙 (X ⟶[V] Y) :=
+enriched_category.comp_id X Y
+
+@[simp, reassoc]
+lemma e_assoc (W X Y Z : C) :
+  (α_ _ _ _).inv ≫ (e_comp V W X Y ⊗ 𝟙 _) ≫ e_comp V W Y Z =
+    (𝟙 _ ⊗ e_comp V X Y Z) ≫ e_comp V W X Z :=
+enriched_category.assoc W X Y Z
+
+section
+variables {V} {W : Type v} [category.{w} W] [monoidal_category W]
+
+/--
+A type synonym for `C`, which should come equipped with a `V`-enriched category structure.
+In a moment we will equip this with the `W`-enriched category structure
+obtained by applying the functor `F : lax_monoidal_functor V W` to each hom object.
+-/
+@[nolint has_inhabited_instance unused_arguments]
+def transport_enrichment (F : lax_monoidal_functor V W) (C : Type u₁) := C
+
+instance (F : lax_monoidal_functor V W) :
+  enriched_category W (transport_enrichment F C) :=
+{ hom := λ (X Y : C), F.obj (X ⟶[V] Y),
+  id := λ (X : C), F.ε ≫ F.map (e_id V X),
+  comp := λ (X Y Z : C), F.μ _ _ ≫ F.map (e_comp V X Y Z),
+  id_comp := λ X Y, begin
+    rw [comp_tensor_id, category.assoc,
+      ←F.to_functor.map_id, F.μ_natural_assoc, F.to_functor.map_id, F.left_unitality_inv_assoc,
+      ←F.to_functor.map_comp, ←F.to_functor.map_comp, e_id_comp, F.to_functor.map_id],
+  end,
+  comp_id := λ X Y, begin
+    rw [id_tensor_comp, category.assoc,
+      ←F.to_functor.map_id, F.μ_natural_assoc, F.to_functor.map_id, F.right_unitality_inv_assoc,
+      ←F.to_functor.map_comp, ←F.to_functor.map_comp, e_comp_id, F.to_functor.map_id],
+  end,
+  assoc := λ P Q R S, begin
+    rw [comp_tensor_id, category.assoc, ←F.to_functor.map_id, F.μ_natural_assoc,
+      F.to_functor.map_id, ←F.associativity_inv_assoc, ←F.to_functor.map_comp,
+      ←F.to_functor.map_comp, e_assoc, id_tensor_comp, category.assoc, ←F.to_functor.map_id,
+      F.μ_natural_assoc, F.to_functor.map_comp],
+  end, }
+
+end
+
+/--
+Construct an honest category from a `Type v`-enriched category.
+-/
+def category_of_enriched_category_Type (C : Type u₁) [𝒞 : enriched_category (Type v) C] :
```

Comment:
Does it make sense to have this as an instance?

Reply:
My preference is to be conservative about adding instances in either direction, until we explore a bit more how this can interoperate with existing enriched categories (honest categories, preadditive categories, linear categories).

Reply:
Sounds good to me!